test/unit/atomic/test-atomic-ref-constructor.cpp

```diff
@@ -0,0 +1,101 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-19, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+///
+/// Source file containing tests for atomic ref constructors (and use of getPointer for verification)
+///
+
+#include <RAJA/RAJA.hpp>
+#include "RAJA_gtest.hpp"
+
+template <typename T>
+void testAtomicDefaultPolConstructors()
+{
+  T * memaddr = nullptr;
+
+  // explicit constructor with memory address
+  RAJA::AtomicRef<T> test1( memaddr );
+
+  ASSERT_EQ( test1.getPointer(), nullptr );
+
+  // ref constructor
+  RAJA::AtomicRef<T> const & reft1 = test1;
+  RAJA::AtomicRef<T> reftest1( reft1 );
+
+  ASSERT_EQ( reftest1.getPointer(), nullptr );
+}
+
+TEST( AtomicRefUnitTest, DefaultPolConstructorsTest )
+{
+  testAtomicDefaultPolConstructors<int>();
+  testAtomicDefaultPolConstructors<float>();
+  testAtomicDefaultPolConstructors<double>();
+}
+
+template <typename T, typename AtomicPolicy>
+void testAtomicBasicConstructors()
+{
+  T * memaddr = nullptr;
+
+  // explicit constructor with memory address
+  RAJA::AtomicRef<T, AtomicPolicy> test1( memaddr );
+
+  ASSERT_EQ( test1.getPointer(), nullptr );
+
+  // ref constructor
+  RAJA::AtomicRef<T, AtomicPolicy> const & reft1 = test1;
+  RAJA::AtomicRef<T, AtomicPolicy> reftest1( reft1 );
+
+  ASSERT_EQ( reftest1.getPointer(), nullptr );
+}
```

Comment:
Should there be a cuda version of this test?

Reply:
Thanks, I'll add a CUDA version.